### ISSUE TITLE
cargo watch failure with `failed to initialize proxy connection: IoFailed`

### ISSUE THREAD

**Issue by audacioustux** [metalbear-co/mirrord#2125]
### Bug Description

if any application fails inside cargo watch, mirrord fails to re-attach with the next iteration of the app.

e.g., `mirrord exec -f .mirrord/mirrord.json -- cargo watch -x 'run --bin billing-api -- --dev'` if the billing-api bin fails in runtime, any next watch trigger will result in this error:
```
[Running 'cargo run --bin billing-api -- --dev']
thread '<unnamed>' panicked at mirrord/layer/src/lib.rs:318:22:
failed to initialize proxy connection: IoFailed(Os { code: 111, kind: ConnectionRefused, message: "Connection refused" })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
mirrord layer setup panicked
```

### Steps to Reproduce

1. create a --bin project with cargo
2. add 
```rust
use std::time::Duration;
use std::thread;

fn main() {
    thread::sleep(Duration::from_millis(4000));

    Err("intentional error").unwrap();
}
```
4. run `mirrord exec -- cargo watch -x run`
5. once the run fails, remove the `Err(...)` line, then save
6. should get `mirrord layer setup panicked` error

### Backtrace

```shell
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: core::result::Result<T,E>::expect
   4: mirrord_layer::layer_start
   5: mirrord_layer::layer_pre_initialization
   6: mirrord_layer::mirrord_layer_entry_point::{{closure}}
   7: std::panicking::try::do_call
   8: __rust_try
   9: std::panicking::try
  10: std::panic::catch_unwind
  11: mirrord_layer::mirrord_layer_entry_point
  12: mirrord_layer::mirrord_layer_entry_point___rust_ctor___ctor::mirrord_layer_entry_point___rust_ctor___ctor
  13: <unknown>
  14: <unknown>
  15: <unknown>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
mirrord layer setup panicked
```

### Your operating system and version

ubuntu

### Local process

cargo-watch

**Comment by aviramha** [comment]
Thanks for reporting this. What mirrord version is being used?

**Comment by audacioustux** [comment]
@aviramha mirrord 3.77.0 ... but looks like it's fixed in 3.78.2.. damn... now it's definitely one of my top 3 favorite tools <3 

anyway, in-case someone search for mirrord + rust integration/example, this might be helpful, so just dropping it here:

```yaml
# https://taskfile.dev/usage/
version: "3"

tasks:
  # cargo check for failing fast on build time errors
  cargo:check:
    cmds:
      - touch .trigger # add .trigger in .gitignore
      - cargo watch -x check -s 'touch .trigger'
    internal: true
  cargo:watch:
    cmd: >-
      mirrord exec -f .mirrord/mirrord.json -- cargo watch 
      --no-vcs-ignores -w .trigger 
      -x 'run'
    internal: true
  cargo:dev:
    deps:
      - cargo:check
      - cargo:watch
```

**Comment by aviramha** [comment]
Awesome :) Kudos to @Razz4780 for a great fix
